.github/workflows/build.yml

```diff
@@ -7,36 +7,68 @@ on:
     branches:
       - main
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
```

Comment:
Does this need github.event.pull_request.number?

Reply:
Probably right.